### PR TITLE
MOSIP-43872 Improved code coverage

### PR DESCRIPTION
**Pull request by kameshsr**
## Summary by CodeRabbit

* **Tests**
  * Widely expanded unit test coverage across the resident service, adding suites for utilities, validators, services and workflows.
  * New tests cover OTP v2 and token/IDA flows, input validation (email/phone/OTP length), masking, localization (status/description/summary), master-data and identity lookups, proxy discard error handling, notification flows and repository interactions.
  * Stronger assertions for error propagation and edge-case handling to improve reliability.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: resident/resident-service/src/test/java/io/mosip/resident/util/CachedIdentityDataUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+ * Unit tests for CachedIdentityDataUtil
+ * @author Kamesh Shekhar Prasad
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CachedIdentityDataUtilTest {
+
+    private CachedIdentityDataUtil cachedIdentityDataUtil;
+
+    @Mock
+    private ResidentServiceRestClient restClientWithSelfTOkenRestTemplate;
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply resident/resident-service/src/test/java/io/mosip/resident/util/CachedIdentityDataUtilTest.java:30]: corrected

File: resident/resident-service/src/main/java/io/mosip/resident/util/CachedIdentityDataUtil.java
Comment (ashok-ksharma) [inline resident/resident-service/src/main/java/io/mosip/resident/util/CachedIdentityDataUtil.java:1]: @kameshsr Remove the code changes because at this moment, we are not expecting application code changes. Only the test case addition is allowed.

**Review by ashok-ksharma — CHANGES_REQUESTED** [review]

File: resident/resident-service/src/main/java/io/mosip/resident/util/CachedIdentityDataUtil.java
Comment (kameshsr) [reply resident/resident-service/src/main/java/io/mosip/resident/util/CachedIdentityDataUtil.java:1]: done

File: resident/resident-service/src/test/java/io/mosip/resident/service/impl/IdAuthServiceTest.java [hunk 2]
```diff
@@ -116,6 +109,12 @@ public class IdAuthServiceTest {
 
     @Mock
     private AvailableClaimUtility availableClaimUtility;
+    
+    @Mock
+    private SessionUserNameUtility sessionUserNameUtility;
+    
+    @Mock
+    private ValidateOtpCharLimit validateOtpCharLimit;
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply resident/resident-service/src/test/java/io/mosip/resident/service/impl/IdAuthServiceTest.java:117]: added usage

File: resident/resident-service/src/test/java/io/mosip/resident/service/impl/EventStatusCodeTest.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+import io.mosip.resident.constant.EventStatus;
+import io.mosip.resident.util.EventStatusBasedOnLangCode;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import reactor.util.function.Tuple2;
+
```
[the remark this answers is not in the document]
Comment (kameshsr) [reply resident/resident-service/src/test/java/io/mosip/resident/service/impl/EventStatusCodeTest.java:13]: its  compiling EventStatusCode is in package io.mosip.resident.service.impl;
EventStatusCodeTest also in same package package io.mosip.resident.service.impl;

[the remark this answers is not in the document]
Comment (kameshsr) [reply resident/resident-service/src/test/java/io/mosip/resident/service/impl/EventStatusCodeTest.java:13]: i will add import also

**Review by JanardhanBS-SyncByte — APPROVED** [review]

**Review by ashok-ksharma — APPROVED** [review]

**Review by Prafulrakhade — APPROVED** [review]
